### PR TITLE
container: do not fail hard, if single containers can't be scraped

### LINKED ISSUE #1473: Windows exporter frequently shows container collector success status as 0
Windows exporter frequently shows container collector success status as 0.
 
```
 C:\>curl http://localhost:9182/metrics | findstr windows_exporter_collector_success
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current 
                                 Dload  Upload   Total   Spent    Left  Speed  
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0# HELP windows_exporter_collector_success windows_exporter: Whether the collector was successful.
# TYPE windows_exporter_collector_success gauge
windows_exporter_collector_success{collector="container"} 0
windows_exporter_collector_success{collector="cpu"} 1
windows_exporter_collector_success{collector="cpu_info"} 1
windows_exporter_collector_success{collector="cs"} 1
windows_exporter_collector_success{collector="logical_disk"} 1
windows_exporter_collector_success{collector="memory"} 1
windows_exporter_collector_success{collector="net"} 1
windows_exporter_collector_success{collector="os"} 1
windows_exporter_collector_success{collector="physical_disk"} 1
windows_exporter_collector_success{collector="process"} 1
windows_exporter_collector_success{collector="service"} 1
windows_exporter_collector_success{collector="system"} 1
windows_exporter_collector_success{collector="textfile"} 1
windows_exporter_collector_success{collector="time"} 1
100 1196k    0 1196k    0     0   897k      0 --:--:--  0:00:01 --:--:--  897k
```

Following error logs which are being observed below  in windows-exporter container logs do not exists containers:

```
  |   |2024-05-09 10:12:08.871 | ts=2024-05-09T10:12:08.871Z caller=prometheus.go:168 level=warn msg="Collection timed out, still waiting for [container]" | 
  |   | 2024-05-09 10:12:16.206 | ts=2024-05-09T10:12:16.197Z caller=container.go:345 level=warn collector=container msg="Failed to collect network stats for container 30d694b1bb2dbe475f97bae02cdb6bb17d659f967c09bde7bea57bba9e354bf6" |  
  |   | 2024-05-09 10:12:16.208 | ts=2024-05-09T10:12:16.207Z caller=container.go:345 level=warn collector=container msg="Failed to collect network stats for container 6ed502afa7dd02f0f04fe7a1819d597ceaffc1ec71b744c6f22fc6e3a5d69ec8" |  
  |   | 2024-05-09 10:12:29.391 | ts=2024-05-09T10:12:29.391Z caller=container.go:246 level=error collector=container msg="err in fetching container Statistics" containerId=80dd707243d5fe6505a5e6da1e37097654d4ab52e895239db5bf3d7799a9fac1 err="container 80dd707243d5fe6505a5e6da1e37097654d4ab52e895239db5bf3d7799a9fac1 encountered an error during hcs::System::Properties: failure in a Windows system call: Element not found. (0x490)" |  
  |   | 2024-05-09 10:12:30.873 | ts=2024-05-09T10:12:30.873Z caller=container.go:345 level=warn collector=container msg="Failed to collect network stats for container 80dd707243d5fe6505a5e6da1e37097654d4ab52e895239db5bf3d7799a9fac1" |  
  |   | 2024-05-09 10:12:59.394 | ts=2024-05-09T10:12:59.394Z caller=container.go:246 level=error collector=container msg="err in fetching container Statistics" containerId=00c560664f3e1046d593bc6ce3a00de51d5060e4ed0ec5d5116498de3ed27dd6 err="container 00c560664f3e1046d593bc6ce3a00de51d5060e4ed0ec5d5116498de3ed27dd6 encountered an error during hcs::System::Properties: failure in a Windows system call: Element not found. (0x490)" |  
  |   | 2024-05-09 10:12:59.442 | ts=2024-05-09T10:12:59.441Z caller=container.go:246 level=error collector=container msg="err in fetching container Statistics" containerId=cc60bc889ae8ea9dfcd2b55f7390b9ca7f9e2735420159dbb6a395769247235a err="container cc60bc889ae8ea9dfcd2b55f7390b9ca7f9e2735420159dbb6a395769247235a encountered an error during hcs::System::Properties: failure in a Windows system call: Element not found. (0x490)"
```

All the data, metrics are visible on Grafana for the windows dashboard. Kindly suggest how to resolve this issue and  what will be causing the container collector status as 0 and the significance of such error logs.

### PR DESCRIPTION
**Pull request by jkroepke**
Fixes #1473

### CODE REVIEW

File: pkg/collector/container/container.go [hunk 1]
```diff
@@ -228,167 +225,159 @@ func (c *Collector) collect(ch chan<- prometheus.Metric) error {
...
-		endpointStats, err := hcsshim.GetHNSEndpointStats(endpoint.Id)
-		if err != nil {
-			_ = level.Warn(c.logger).Log("msg", "Failed to collect network stats for interface "+endpoint.Id, "err", err)
-			continue
+	defer func() {
+		if container == nil {
+			return
 		}
 
-		for _, containerId := range endpoint.SharedContainers {
```
Comment (jsturtevant) [inline pkg/collector/container/container.go:345]: you've dropped this,  It is tempting to use the `.network` to collect the stats but you can't do that with containerd's hns v2.  `.network` only works on v1 hns apis.  
see https://github.com/prometheus-community/windows_exporter/commit/9adefdceeb6a6467bda2222f124e72f6a8081e36 and https://github.com/prometheus-community/windows_exporter/pull/1218

Comment (jkroepke) [reply pkg/collector/container/container.go:345]: Missing knowledge and missing comments, thanks for the hints!

Comment (jkroepke) [reply pkg/collector/container/container.go:345]: I have re-add this, thanks a lot!

**Comment by jsturtevant** [comment]
This looks like it does the correct thing, can we run tests and see how its working?  I don't have an environment to test things out right now but if someone does that would add some additional validation

**Comment by jkroepke** [comment]
Collected tested with `ghcr.io/jkroepke/windows-exporter:pr1561` on a Windows Server 2022 AKS node

**Pull request merged**